sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/RecognizePiiEntityAsyncClient.java

```diff
@@ -0,0 +1,215 @@
+     * @param context Additional context that is passed through the Http pipeline during the service call.
+     *
+     * @return A mono {@link Response} that contains {@link RecognizePiiEntitiesResultCollection}.
+     */
+    Mono<Response<RecognizePiiEntitiesResultCollection>> recognizePiiEntitiesBatchWithContext(
+        Iterable<TextDocumentInput> documents, TextAnalyticsRequestOptions options, Context context) {
+        try {
+            inputDocumentsValidation(documents);
+            return getRecognizePiiEntitiesResponse(documents, options, context);
+        } catch (RuntimeException ex) {
+            return monoError(logger, ex);
+        }
+    }
+
+    /**
+     * Helper method to convert the service response of {@link EntitiesResult} to {@link Response} which contains
+     * {@link RecognizePiiEntitiesResultCollection}.
+     *
+     * @param response the {@link Response} of {@link EntitiesResult} returned by the service.
+     *
+     * @return A {@link Response} that contains {@link RecognizePiiEntitiesResultCollection}.
+     */
+    private Response<RecognizePiiEntitiesResultCollection> toRecognizePiiEntitiesResultCollectionResponse(
+        final Response<EntitiesResult> response) {
+        final EntitiesResult entitiesResult = response.getValue();
+        // List of documents results
+        final List<RecognizePiiEntitiesResult> recognizeEntitiesResults = new ArrayList<>();
+        entitiesResult.getDocuments().forEach(documentEntities ->
+            recognizeEntitiesResults.add(new RecognizePiiEntitiesResult(
+                documentEntities.getId(),
+                documentEntities.getStatistics() == null ? null
+                    : toTextDocumentStatistics(documentEntities.getStatistics()),
+                null,
+                new PiiEntityCollection(
+                    new IterableStream<>(documentEntities.getEntities().stream().map(entity ->
+                        new PiiEntity(entity.getText(), EntityCategory.fromString(entity.getCategory()),
+                            entity.getSubcategory(), entity.getOffset(), entity.getLength(),
+                            entity.getConfidenceScore()))
+                        .collect(Collectors.toList())),
+                    new IterableStream<>(documentEntities.getWarnings().stream()
+                        .map(warning -> {
+                            final WarningCodeValue warningCodeValue = warning.getCode();
+                            return new TextAnalyticsWarning(
+                                WarningCode.fromString(warningCodeValue == null ? null : warningCodeValue.toString()),
+                                warning.getMessage());
+                        }).collect(Collectors.toList())))
+            )));
+        // Document errors
+        entitiesResult.getErrors().forEach(documentError -> {
+            /*
+             *  TODO: Remove this after service update to throw exception.
```

Comment:
we can remove this now :) https://github.com/Azure/azure-sdk-for-net/issues/11946